desk/app/channels-server.hoon

```diff
@@ -92,27 +92,149 @@
   =?  old  ?=(%3 -.old)  (state-3-to-4 old)
   =?  old  ?=(%4 -.old)  (state-4-to-5 old)
   =?  old  ?=(%5 -.old)  (state-5-to-6 old)
-  ?>  ?=(%6 -.old)
+  =?  old  ?=(%6 -.old)  (state-6-to-7 old)
+  ?>  ?=(%7 -.old)
   =.  state  old
   inflate-io
   ::
-  +$  versioned-state  $%(state-6 state-5 state-4 state-3 state-2 state-1 state-0)
-  +$  state-6  current-state
+  +$  versioned-state  $%(state-7 state-6 state-5 state-4 state-3 state-2 state-1 state-0)
+  +$  state-7  current-state
+  +$  state-6
+    $:  %6
+      =v-channels:v7:old:c
+      =pimp:imp
+    ==
   +$  state-5
     $:  %5
         =v-channels:v6:old:c
         =pimp:imp
     ==
+  ++  state-6-to-7
+    |=  s=state-6
+    ^-  state-7
+    s(- %7, v-channels (v-channels-6-to-7 v-channels.s))
+  ++  v-channels-6-to-7
+    |=  vc=v-channels:v7:old:c
+    ^-  v-channels:c
+    %-  ~(run by vc)
+    |=  v=v-channel:v7:old:c
+    ^-  v-channel:c
+    =/  [log=log:v-channel:c mod=(map id-post:c time)]
+      (log-6-to-7 log.v)
+    =/  [count=@ud =v-posts:c]
+      (v-posts-6-to-7 posts.v mod)
```

Comment:
@arthyn are we fine with only sequence-number-ing the top-level posts? The purpose here is to make it easier for the client to detect & fetch missing chunks of backlog, so I'm assuming it's Probably Fine™, but maybe there's still concerns around large numbers of replies or something like that.

Reply:
yeah I think it's fine for top level only